spec/features/nav_spec.rb

```diff
@@ -1,29 +1,83 @@
 require 'rails_helper'
 
 RSpec.describe "navigation bar", type: :feature do
-  it 'visitor navigation' do
-    visit items_path
+  context "as a visitor" do
+    it 'has working links' do
+      visit items_path
 
-    within("#nav") do
-      click_link "Home"
-      expect(current_path).to eq(root_path)
+      within("nav") do
+        click_link "Home"
+        expect(current_path).to eq(root_path)
 
-      click_link "Browse All Items"
-      expect(current_path).to eq(items_path)
+        click_link "Browse All Items"
+        expect(current_path).to eq(items_path)
 
-      # click_link "Browse All Merchants"
-      # expect(current_path).to eq("/merchants")
+        click_link "Browse All Merchants"
+        expect(current_path).to eq(merchants_path)
 
-      # click_link "My Cart"
-      # expect(current_path).to eq("/cart")
+        # to-do:
+        # click_link "My Cart"
+        # expect(current_path).to eq("/cart")
 
-      click_link "Login"
-      expect(current_path).to eq(login_path)
+        click_link "Login"
+        expect(current_path).to eq(login_path)
 
-      click_link "Register"
-      expect(current_path).to eq("/register")
+        click_link "Register"
+        expect(current_path).to eq(register_path)
+
+        expect(page).to_not have_content "Logged in as"
+      end
+    end
+
+    it "Next to the shopping cart link I see a count of the items in my cart"
+    # to-do
+  end
+
+  context "as a user" do
+    before(:each) do
```

Comment:
If you remove before :each and stub the login inside the example, you can check that the login message is no longer there.